mozregression/launchers.py

```diff
@@ -173,6 +177,39 @@ class MozRunnerLauncher(Launcher):
     app_name = "undefined"
     binary = None
 
+    @staticmethod
+    def _codesign_verify(appdir):
+        """Calls `codesign` to verify signature, and returns state."""
+        if mozinfo.os != "mac":
+            raise Exception("_codesign_verify should only be called on macOS.")
+
+        try:
+            output = check_output(["codesign", "-v", appdir], stderr=STDOUT)
+        except CalledProcessError as e:
+            output = e.output
+            exit_code = e.returncode
+        else:
+            exit_code = 0
+
+        LOG.debug(f"codesign verify exit code: {exit_code}")
+
+        if exit_code == 0:
+            return CodesignResult.PASS
+        elif exit_code == 1:
+            if b"code object is not signed at all" in output:
+                return CodesignResult.UNSIGNED
+            else:
+                return CodesignResult.INVALID
+        # Remaining codes normally mean the command was called with incorrect arguments.
+        return CodesignResult.OTHER
+
+    @staticmethod
+    def _codesign_sign(appdir):
+        """Calls `codesign` to sign `appdir` with ad-hoc identity."""
+        if mozinfo.os != "mac":
+            raise Exception("_codesign_sign should only be called on macOS.")
+        return call(["codesign", "--force", "--deep", "--sign", "-", appdir])
```

Comment:
are there risks of different versions of `codesign` having different arguments and/or different output messages which could affect this?  Maybe indicate with a comment which os version and codesign version you tested with?

Reply:
Good point, will add a comment.

Reply:
I checked the documentation for all versions (going back to Mojave - https://www.unix.com/man-page/mojave/1/codesign/) and the arguments appear to all be the same. In terms of the output message, I was only able to check macOS 11, 12, and 13 which are the only ones I have access to. The output message is the same in all.